### PR TITLE
Company page

### PR DESCRIPTION
**Pull request by akbarjorayev**
APIs:
```
https://api.bandla.uz/api/user-panel/company/create
```
```
https://api.bandla.uz/api/user-panel/company/find
```
Create company page

### CODE REVIEW

**Review by nazarovctrl — CHANGES_REQUESTED** [review]
Kalla qoyibsanu updatePhoto ni ichini yaxshilab qara keremas kodi copy paste qigansan

File: src/pages/Company.js [hunk 1]
```diff
@@ -1,7 +1,192 @@
...
+      "Profil rasmi o'chirildi",
+      "Profile rasmini o'chirishda xatolik ro'y berdi"
+    );
+  };
+
+  const navigateToLogin = () => {
+    window.location.href = 'login';
+  };
+
+  const updatePhoto = async (photoUrl, successMessage, errorMessage) => {
```
[the remark this answers is not in the document]
Comment (akbarjorayev) [reply src/pages/Company.js:32]: Got it

[the remark this answers is not in the document]
Comment (akbarjorayev) [reply src/pages/Company.js:32]: API ni ishlatib korsatib ber bro

File: src/components/Input.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useRef, useState } from 'react';
```
Comment (nazarovctrl) [inline src/components/Input.js:1]: nega alohida Input ishlatvosan, tayyori yogmi?  Bu Input  Profile formasidagi inputdan qanaqa afzalligi bor

File: src/pages/App.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ import Profile from './Profile';
 import Login from './Login';
 import Logout from './Logout';
 import About from './About';
+import Company from './Company';
 import WrongPath from './WrongPath';
```
Comment (nazarovctrl) [inline src/pages/App.js:9]: wrongPath importiniyam oxiriga qoyvor

File: src/pages/Company.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
+import { useEffect, useRef, useState } from 'react';
+import Input from '../components/Input';
+import { Avatar } from '@mui/material';
+import Alert from '../components/Alert';
+import Request from '../util/Request';
+import * as IMG from '../util/Image.js';
+
+import { toast } from 'react-toastify';
```
Comment (nazarovctrl) [inline src/pages/Company.js:8]: nimabu nogironmi ajratib qoyibsan😂

File: src/pages/Company.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+export default function Company() {
+  const companyName = useRef();
+  const companyAddress = useRef();
+  const fileInput = useRef(null);
+  const [photoUrl, setPhotoUrl] = useState('');
+  const [showAlertPic, setShowAlertPic] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+  const maxPhotoSize = 1024 * 1024 * 6; //KB
+  const FINDURL =
```
Comment (nazarovctrl) [inline src/pages/Company.js:20]: nega kere bu?

**Review by nazarovctrl — CHANGES_REQUESTED** [review]

File: src/components/Input.js (hunk 2 above)
Comment (akbarjorayev) [reply src/components/Input.js:1]: Bu input ni san xohlaganindaqa style qoyolisan

File: src/pages/App.js (hunk 3 above)
Comment (akbarjorayev) [reply src/pages/App.js:9]: ok

File: src/pages/Company.js (hunk 4 above)
Comment (akbarjorayev) [reply src/pages/Company.js:8]: man emas, bu code ni man copy paste qilganamn

File: src/pages/Company.js (hunk 5 above)
Comment (akbarjorayev) [reply src/pages/Company.js:20]: nima FINDURL ni, agar shu bosa ketb qolbdi. keremas

File: src/pages/Company.js (hunk 4 above)
Comment (nazarovctrl) [reply src/pages/Company.js:8]: Hop copy paste qilasanmi boshqa narsa qilasanmi bu codeni yozvotgan ozin shunga togrlab yoz

Comment (nazarovctrl) [reply src/pages/Company.js:8]: togrilanmagan!

File: src/pages/Company.js (hunk 5 above)
Comment (nazarovctrl) [reply src/pages/Company.js:20]: togrilanmagan!

File: src/pages/Company.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+  }
+
+  async function createCompany() {
+    const companyData = {
+      name: companyName.current.value,
+      address: companyAddress.current.value,
+      photoUrl,
+    };
+
+    const something = await Request(
```
Comment (nazarovctrl) [inline src/pages/Company.js:93]: try catch qani?

**Review by nazarovctrl — CHANGES_REQUESTED** [review]

File: src/pages/App.js (hunk 3 above)
Comment (nazarovctrl) [reply src/pages/App.js:8]: eng oxirigia qoyish keredi, company ni pasigamas

**Review by nazarovctrl — CHANGES_REQUESTED** [review]

**Comment by akbarjorayev** [comment]
Show **Status**!

File: src/components/Input.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import CloseIcon from '@mui/icons-material/Close';
+
```
Comment (nazarovctrl) [inline src/components/Input.js:3]: nega kere bu bosh qator? ochirib tasha

File: src/components/Navbar.js [hunk 8]
```diff
@@ -61,8 +62,9 @@ function Navbar() {
   const handleClick = (event) => {
     setAnchorEl(event.currentTarget);
   };
-  const handleClose = () => {
+  const handleClose = (path) => {
     setAnchorEl(null);
+    if (typeof path === 'string') window.location.href = path;
```
Comment (nazarovctrl) [inline src/components/Navbar.js:67]: nega if kere?  o'zi bu handleClose ga faqat string  kevotganini usizam bilib turibsanu

File: src/pages/Company.js [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+          'get',
+          null,
+          null,
+          true,
+          null,
+          navigateToLogin
+        );
+
+        setCompany(response);
+        setPhotoUrl(response.data.data.photoUrl);
```
Comment (nazarovctrl) [inline src/pages/Company.js:33]: bu logika setCompany() ichida boladigan qilish kere

File: src/pages/Company.js [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+    try {
+      const response = await Request(
+        'https://api.bandla.uz/api/user-panel/company/create',
+        'post',
+        null,
+        companyData,
+        true,
+        navigateToLogin
+      );
+
```
Comment (nazarovctrl) [inline src/pages/Company.js:120]: response dagi danniy ishlatilmagan! for more information -> telegram chat

**Review by nazarovctrl — CHANGES_REQUESTED** [review]

**Review by akbarjorayev — COMMENTED** [review]
Review this @nazarovctrl PLS

File: src/pages/Company.js [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+              disabled
+            />
+          )}
+          {!company?.data?.data?.status && (
+            <Input ref={companyAddress} type="text" label="Manzil" />
+          )}
+          {!company?.data?.data?.status && (
+            <button onClick={createCompany}>Tasdiqlash</button>
+          )}
+          {company?.data?.data?.status && <button>Created</button>}
```
Comment (nazarovctrl) [inline src/pages/Company.js]: xato status qiymatini korsatish kerak. hozirgi code da status bolsa faqat Created korinadi

File: src/pages/Company.js [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+      const response = await Request(
+        'https://api.bandla.uz/api/user-panel/company/create',
+        'post',
+        null,
+        companyData,
+        true,
+        navigateToLogin
+      );
+
+      setCompany(response);
```
Comment (nazarovctrl) [inline src/pages/Company.js:119]: xato setCopmany ichiga faqat company malutmoalrini berish kere hozi resonse ichida request jonatishdagi har xil balo battar musr malumotla bor
```suggestion
      setCompany(response.data.data);
```

File: src/pages/Company.js [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+        'https://api.bandla.uz/api/file-store',
+        'post',
+        null,
+        formData,
+        true,
+        navigateToLogin
+      );
+
+      setShowAlertPic(false);
+      setCompany(response);
```
Comment (nazarovctrl) [inline src/pages/Company.js:75]: xato file-store api dan company ga oid malumto kemidi lekin setCompany qilinvoti, 
Yechim:
company ichidagi url ni response.data.data.url ga tenglab qoyish kerak
company.url = response.data.data.url

File: src/pages/Company.js [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+          'https://api.bandla.uz/api/user-panel/company/find',
+          'get',
+          null,
+          null,
+          true,
+          null,
+          navigateToLogin
+        );
+
+        setCompany(response);
```
Comment (nazarovctrl) [inline src/pages/Company.js:31]:
```suggestion
        setCompany(response.data.data);
```

**Review by nazarovctrl — CHANGES_REQUESTED** [review]

**Review by akbarjorayev — COMMENTED** [review]
Status **fixed** @nazarovctrl

**Review by akbarjorayev — COMMENTED** [review]
I think **All** comments were **resolved**! @nazarovctrl

File: src/pages/Company.js [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+              >
+                Yopish
+              </button>
+              <button buttonkey="true" onClick={changePhotoClick}>
+                Saqlash
+              </button>
+            </Alert>
+          )}
+        </div>
+        <div className="inputs">
```
Comment (nazarovctrl) [inline src/pages/Company.js:195]: Inputlarni  disabled bolishini yaxshilash kerak

**Review by nazarovctrl — DISMISSED** [review]

File: src/pages/Company.js (hunk 15 above)
Comment (akbarjorayev) [reply src/pages/Company.js:195]: Qilomoman manmcha

File: src/pages/Company.js [hunk 16]
```diff
@@ -0,0 +1,233 @@
...
+              <Input
+                ref={companyAddress}
+                type="text"
+                label="Manzil"
+                value={company?.address}
+                disabled
+              />
+            </>
+          )}
+          {!company?.id && (
```
Comment (akbarjorayev) [inline src/pages/Company.js:217]: They actually **work** @nazarovctrl

**Review by akbarjorayev — COMMENTED** [review]
Good! But 2 Inputs!

File: src/pages/Company.js (hunk 15 above)
Comment (akbarjorayev) [reply src/pages/Company.js:214]: boldi @nazarovctrl

**Review by akbarjorayev — COMMENTED** [review]
Good 👍

**Review by nazarovctrl — APPROVED** [review]
LGTM

**Pull request merged**